neard/src/cli.rs

```diff
@@ -258,6 +258,9 @@ impl RunCmd {
         let sys = actix::System::new();
         sys.block_on(async move {
             nearcore::start_with_config(home_dir, near_config);
+            tokio::signal::ctrl_c().await.unwrap();
+            info!("Got Ctrl+C, stopping");
+            actix::System::current().stop();
         });
```

Comment:
For posterity, while this is undoubtedly a big improvement, I think that this is not the right end-game logic. 

Proper shutdown should be a three-phase process.

* first, you *request* the system to cancel. 
* second, you *wait* for the system to actually stop.
* finally, if the system didn't stop within a time limit, you abruptly terminate it.

For networked services, such graceful shutdown means that all currently-pending requests are finished, but new connections and new requests are not accepted.

The `actix::System::current().stop();` I think goes directly to abrupt shutdown. So, if the node is in the middle of sending an HTTP response, it won't be send in full.

OTOH, there's an argument that, in a P2P byzantine network, graceful shutdown is a misfeature, as it adds a happy path, while the code should always be robust for all sad paths.